### PR TITLE
Fixing missing json messages while file redirect

### PR DESCRIPTION
**Pull request by karthikprabhuvinod**
Signed-off-by: Karthik Prabhu Vinod <karthik.prabhu.vinod@intel.com>

### CODE REVIEW

File: src/lib/progress.c [hunk 1]
```diff
@@ -153,20 +153,20 @@ static void progress_spinner_print_end(void)
...
-	if (log_get_level() == LOG_DEBUG || !isatty(fileno(stdout))) {
+	if (log_get_level() == LOG_DEBUG) {
 		return;
 	}
 
 	if (status_flag) {
 		if (progress_function) {
 			progress_report(-1, 100);
-		} else {
+		} else if (isatty(fileno(stdout))) {
```
Comment (otaviobp) [inline src/lib/progress.c]: can you do the same on the case of --debug?

if you run

    swupd diagnose --debug --json-output

we should print the -1 too

Comment (karthikprabhuvinod) [reply src/lib/progress.c]: Done

**Comment by karthikprabhuvinod** [comment]
@otaviobp : Fixed it

File: src/lib/progress.c [hunk 2]
```diff
@@ -153,20 +153,20 @@ static void progress_spinner_print_end(void)
...
 {
-	if (log_get_level() == LOG_DEBUG || !isatty(fileno(stdout))) {
-		return;
-	}
-
 	if (status_flag) {
 		if (progress_function) {
 			progress_report(-1, 100);
-		} else {
+		} else if (log_get_level() == LOG_DEBUG) {
```
Comment (otaviobp) [inline src/lib/progress.c]: It works now, but it's clearer if you use the same if for log_debug and isatty. So you don't need a return here.

Comment (karthikprabhuvinod) [reply src/lib/progress.c]: Done

**Comment by karthikprabhuvinod** [comment]
@castulo : Please take a look and let me know if that makes sense
@otaviobp : Please take a look

**Pull request merged**